### PR TITLE
Add package gccxml

### PR DESCRIPTION
**Pull request by healther**
There is unfortunately no buildable release (the 0.6.x fails with a CMake error). I tagged the last change to code (i.e. ignoring the README changes from a couple of months ago)

### CODE REVIEW

File: var/spack/repos/builtin/packages/gccxml/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    homepage = "http://gccxml.github.io"
+    url      = "https://github.com/gccxml/gccxml/archive/v0.6.x.tar.gz"
+
+    version('develop', git='https://github.com/gccxml/gccxml.git', branch='master')
+    version('latest', git='https://github.com/gccxml/gccxml.git', 
+            commit='3afa8ba5be6866e603dcabe80aff79856b558e24', preferred=True)
+
+    if sys.platform == 'darwin':
+        patch('darwin-gcc.patch', when="%gcc")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gccxml/package.py]: You should be able to use `when='%gcc platform=darwin'`.

Comment (healther) [inline var/spack/repos/builtin/packages/gccxml/package.py]: I wasn't sure about the exact syntax, could you point me to the documentation for it? I failed to find it on my own, and worked around it by grepping in other ```package.py```'s...

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gccxml/package.py]: There is no documentation 😆 @tgamblin

File: var/spack/repos/builtin/packages/gccxml/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import sys
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gccxml/package.py]: You can remove this import statement now that it isn't needed anymore.

File: var/spack/repos/builtin/packages/gccxml/package.py (hunk 1 above)
Comment (healther) [inline var/spack/repos/builtin/packages/gccxml/package.py]: Okay... at least I didn't fail at finding it, I guess

**Pull request merged**